site/docs/hosting/firebase.md

```diff
@@ -0,0 +1,43 @@
+# Hosting: Firebase Functions
+
+This tutorial will guide you on how to deploy your bot to [firebase](https://firebase.google.com/) by using [Firebase Serverless Functions](https://firebase.google.com/docs/functions), assuming that you already have a firebase account.
+
+## Setup
+
+In order to deploy a firebase function, you will need to create a firebase account, install their CLI, and create a firebase project.
+You should first [follow their documentation](https://firebase.google.com/docs/functions/get-started) to set things up.
+
+> You can ommit steps 4 to 6 since they are just examples. It is usefull though to set up the emulators (step 7) to test your code locally.
+
+## Preparing Your Code
+
+You can use this short example bot as a starting point
+
+```ts
+import * as functions from "firebase-functions";
+import {webhookCallback, Bot} from "grammy";
+
+const bot = new Bot(<BOT_TOKEN>);
+
+bot.command("start", (ctx) => ctx.reply("Welcome! Up and running."));
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+//you can trigger your function on https://<domain>/<firebase-projectname>/helloworld
```

Comment:
Please elaborate on what you mean by trigger the function.

Also, the links here look different from the ones in the Deploying section: `https://<domain>/<firebase-projectname>/helloworld` vs `<https://<REGION>.<MY_PROJECT>.cloudfunctions.net/helloworld>`. Are both links unrelated?

Reply:
The url structure for the emulated firebase is different from the deployed one for some reason.
I specified in the comment that this is for development now